scripts/bulk_capture.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+
+# Help message
+function show_help {
+    echo "Usage: $0 [--help]"
+    echo ""
+    echo "  --help       Print this message"
+    exit 1
+}
+
+# Check if --help is passed
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --help) show_help ;;
+        *) echo "Unknown parameter passed: $1"; show_help ;;
+    esac
+    shift
+done
+
+# File containing the ranges
+range_file="./range.csv"
+
+# Generate a unique temporary directory name
+uuid=$(uuidgen)
+remote_dir="/tmp/replays-$uuid"
+
+# Check if the range file exists
+if [ ! -f "$range_file" ]; then
+    echo "Error: range file $range_file does not exist"
+    exit 1
+fi
+
+# Initialize the full command
+full_cmd="mkdir -p $remote_dir; "
+
+# Variables to track the very first and very last ranges
+very_first=""
+very_last=""
+
+# Read the range file and process each range
+while IFS= read -r line || [[ -n "$line" ]]; do
+    # Debug: Show the current line being processed
+    echo "Debug: Current line: '$line'"
+
+    # Skip empty lines
+    if [ -z "$line" ]; then
+        echo "Debug: Skipping empty line"
+        continue
+    fi
+
+    # Extract start and end from the line
+    start=$(echo "$line" | cut -d'-' -f1)
+    end=$(echo "$line" | cut -d'-' -f2)
+
+    # Debug: Show extracted start and end values
+    echo "Debug: Extracted start: '$start', end: '$end'"
+
+    # Check if start and end are properly extracted
+    if [ -z "$start" ] || [ -z "$end" ]; then
+        echo "Debug: Skipping invalid line: '$line'"
+        continue
+    fi
+
+    # Track the very first and very last ranges
+    if [ -z "$very_first" ]; then
+        very_first="$start"
+    fi
+    very_last="$end"
+
+    # Set remote filename
+    remote_filename="$remote_dir/$start-$end.json.gz"
+
+    # Validate range
+    if [ "$start" -gt "$end" ]; then
+        echo "Invalid range: $start - $end"
+        continue
+    fi
+
+    echo "Capturing conflation $start - $end"
+    echo "Writing replay to \`$remote_filename\`"
+
+    # Form payload
+    payload=$(cat <<EOF
+{
+   "jsonrpc":"2.0",
+   "method":"linea_captureConflation",
+   "params":["$start", "$end"], "id":"1"
+}
+EOF
+    )
+
+    # Append the command to full_cmd
+    full_cmd+="curl -X POST 'http://localhost:8545' --data '$payload' | jq '.result.capture' -r | jq . | gzip > $remote_filename; echo 'Captured $start-$end'; sleep 3; "
+done < "$range_file"
+
+# Set the compressed file name based on the very first and very last ranges
+compressed_file="/tmp/replays_${very_first}_${very_last}.tar.gz"
+local_compressed_file="replays_${very_first}_${very_last}.tar.gz"
```

Comment:
Those seem like slightly odd names for a collection of conflated ranges, maybe it would be a better convention to call it something like

```bash
compressed_file="/tmp/replays_${number_of_ranges}_conflations_between_${very_first}_and_${very_last}.tar.gz"
local_compressed_file="replays_${number_of_ranges}_conflations_between_${very_first}_and_${very_last}.tar.gz"
```

Where number_of_ranges would be the number of rows in the .csv file. Or something similar.

Reply:
I got your point. I originally used these ranges so avoid duplicate names, but now we had that uuid.  I think we can add more context to the compressed file name. Let me fix this.

Reply:
Updated